tortoise/backends/base/executor.py

```diff
@@ -318,15 +345,23 @@ def _make_prefetch_queries(self) -> None:
     async def _do_prefetch(self, instance_id_list: list, field: str, related_query) -> list:
         if field in self.model._meta.backward_fk_fields:
             return await self._prefetch_reverse_relation(instance_id_list, field, related_query)
+
+        if field in self.model._meta.backward_o2o_fields:
+            return await self._prefetch_reverse_o2o_relation(instance_id_list, field, related_query)
+
         if field in self.model._meta.m2m_fields:
             return await self._prefetch_m2m_relation(instance_id_list, field, related_query)
         return await self._prefetch_direct_relation(instance_id_list, field, related_query)
 
     async def _execute_prefetch_queries(self, instance_list: list) -> list:
         if instance_list and (self.prefetch_map or self._prefetch_queries):
             self._make_prefetch_queries()
-            for field, related_query in self._prefetch_queries.items():
-                await self._do_prefetch(instance_list, field, related_query)
+            prefetch_tasks = [
+                self._do_prefetch(instance_list, field, related_query)
+                for field, related_query in self._prefetch_queries.items()
+            ]
+            await asyncio.gather(*prefetch_tasks)
```

Comment:
:+1: 
Yes, since we are now properly concurrency safe, we should probably do more I/O concurrently instead of in loops